hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/SCMDBMetaDataInitializationTask.java

```diff
@@ -0,0 +1,176 @@
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(SCMDBMetaDataInitializationTask.class);
+
+  private OzoneConfiguration configuration;
+  private ReconScmMetadataManager scmMetadataManager;
+  private ReconContainerMetadataManager reconContainerMetadataManager;
+  private Map<String, SCMMetaDataTableHandler> tableHandlers;
+  private Map<HddsProtos.LifeCycleState, Long> containerCountMap;
+
+  @Inject
+  public SCMDBMetaDataInitializationTask(ReconScmMetadataManager scmMetadataManager,
+                                         ReconContainerMetadataManager reconContainerMetadataManager,
+                                         OzoneConfiguration configuration) throws IOException {
+    this.scmMetadataManager = scmMetadataManager;
+    this.reconContainerMetadataManager = reconContainerMetadataManager;
+    this.configuration = configuration;
+
+    containerCountMap = new HashMap<>();
+
+    // Initialize table handlers
+    tableHandlers = new HashMap<>();
+    tableHandlers.put(ReconScmMetadataManager.CONTAINERS_TABLE,
+        new ContainersInfoHandler(reconContainerMetadataManager, scmMetadataManager, containerCountMap));
+    tableHandlers.put(ReconScmMetadataManager.PIPELINES_TABLE, new PipelineInfoHandler(scmMetadataManager));
+    tableHandlers.put(ReconScmMetadataManager.SEQUENCE_ID_TABLE, new SeqIdGenInfoHandler(scmMetadataManager));
+  }
+
+  /**
+   * Read container table data and compute stats like DELETED container count,
+   * number of blocks per container etc.
+   */
+  @Override
+  public Pair<String, Boolean> reprocess(ReconScmMetadataManager reconScmMetadataManager) throws IOException {
+    for (String tableName : getTaskTables()) {
+      Table table = scmMetadataManager.getTable(tableName);
+      if (table == null) {
+        LOG.error("Table " + tableName + " not found in SCM Metadata.");
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+      if (tableHandlers.containsKey(tableName)) {
+        tableHandlers.get(tableName).reprocess(reconScmMetadataManager);
+      }
+    }
+    LOG.info("Completed a 'reprocess' run of SCMDBMetaDataInitializationTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+  @Override
+  public String getTaskName() {
+    return "SCMDBMetaDataInitializationTask";
+  }
+
+  public Collection<String> getTaskTables() throws IOException {
+    List<String> taskTables = new ArrayList<>();
+    taskTables.add(scmMetadataManager.getContainerTable().getName());
+    taskTables.add(scmMetadataManager.getPipelineTable().getName());
+    taskTables.add(scmMetadataManager.getSequenceIdTable().getName());
+    return taskTables;
+  }
+
+  @Override
+  public Pair<String, Boolean> process(RocksDBUpdateEventBatch events) {
+    Iterator<RocksDBUpdateEvent> eventIterator = events.getIterator();
+    int eventCount = 0;
+    Collection<String> taskTables = Collections.EMPTY_SET;
```

Comment:
Instead of using `Collections.EMPTY_SET,` which is raw-typed, prefer using Collections.emptySet(), which is type-safe and can be assigned directly to a variable of type Collection<String> without casting: